library_generation/generate_composed_library.sh

```diff
@@ -0,0 +1,107 @@
+#!/bin/bash
+# This script allows generation of libraries that are composed of more than one
+# service version. It is achieved by calling `generate_library.sh` without
+# postprocessing for all service versions and then calling
+# postprocess_library.sh at the end, once all libraries are ready.
+#
+# Arguments
+# --generation_queries: a single string of key-value groups separated by a
+# pipe |. They key-value groups are in the form of `key=value` and will
+# be converted to an argument to generate_library.sh (`--key value`).
+#   example: "proto_path=google/cloud/asset/v1,destination_path=google-cloud-asset-v1-java,(...)|proto_path=google/cloud/asset/v1p2beta5,destination_path=google-cloud-asset-v1-java,(...)"
+#   In this case, generate_library.sh will be called once with
+#     generate_library.sh --proto_path google/cloud/asset/v1 --destination_path google-cloud-asset-v1-java
+#     and once with
+#     generate_library.sh --proto_path google/cloud/asset/v1p2bet5 --destination_path google-cloud-asset-v1p2beta5-java
+# --versions_file: a file of newline-separated version strings in the form "module:released-version:current-version". The versions will be applied to the pom.xml files and readmes
+#
+# Prerequisites
+# - Needs an `output` folder at the location of the calling shell. This folder
+# is automatically detected by `generate_library.sh` and this script ensures it
+# contains the necessary folders and files, specifically:
+#   - A "google" folder found in the googleapis repository
+#   - A "grafeas" folder found in the googleapis repository
+# Note: googleapis repo is found in https://github.com/googleapis/googleapis.
+
+set -xeo pipefail
+
+
+script_dir=$(dirname "$(readlink -f "$0")")
+library_generation_dir="${script_dir}"
+source "${script_dir}/utilities.sh"
+output_folder=$(get_output_folder)
+mkdir -p "${output_folder}"
+
+while [[ $# -gt 0 ]]; do
+key="$1"
+case $key in
+  -g|--generation_queries)
+    generation_queries="$2"
+    ;;
+  -r|--repository_path)
+    repository_path="$2"
+    ;;
+  -v|--versions_file)
+    versions_file="$2"
+    ;;
+  -p|--enable_postprocessing)
+    enable_postprocessing="$2"
+    ;;
+  *)
+    echo "Invalid option: [$1]"
+    exit 1
+    ;;
+esac
+shift
+shift # past argument or value
+done
+
+
+pushd "${output_folder}"
+if [[ "${repository_path}" == google-cloud-java/* ]]; then
```

Comment:
I don't think we want to hardcode the check here, the script would fail if we rename the repo. It should be a parameter passed in from the caller. e.g. the [destination_path](https://github.com/googleapis/sdk-platform-java/pull/2337/files#diff-3153a678f540ff9e6f64cf1a0f7e8ac815dfdfeae73bb9998b5d9ceb2dfdea96R12) in the yaml config file.

Reply:
I agree it would be nice to not rely on heuristics like this one. Just like in the comment below, the config yaml may make this easier and deterministic

Reply:
I'm leaving this as a pending item once we merge the config yaml PR